config/config_test.go

```diff
@@ -0,0 +1,284 @@
+			args: args{
+				confFilePath: "./test-config.yaml",
+			},
+			wantErr: false,
+		},
+		{
+			name: "LoadConfEmptyFileName",
+			args: args{
+				confFilePath: "",
+			},
+			wantErr: false,
+		},
+		{
+			name: "LoadConfBadYamlFile",
+			args: args{
+				confFilePath: "./test-bad-config.yaml",
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := LoadConf(tt.args.confFilePath); (err != nil) != tt.wantErr {
+				t.Errorf("LoadConf() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+	DeleteFile()
+}
+
+func Test_getConfValueByStringField(t *testing.T) {
+	type args struct {
+		conf  Configuration
+		field string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "KeyPresentInYaml",
+			args: args{
+				conf: Configuration{
+					Environment: "minikube",
+				},
+				field: "Environment",
+			},
+			want: "minikube",
+		},
+		{
+			name: "KeyNotPresentInYaml",
+			args: args{
+				conf: Configuration{
+					Environment: "minikube",
+				},
+				field: "environment",
+			},
+			want: "<invalid Value>",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := getConfValueByStringField(tt.args.conf, tt.args.field); got != tt.want {
+				fmt.Println(got)
+				t.Errorf("getConfValueByStringField() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestGetDefaultValueByStringField(t *testing.T) {
+	type args struct {
+		field string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "FieldPresentInYaml",
+			args: args{
+				field: "Environment",
+			},
+			want: "minikube",
+		},
+		{
+			name: "FieldNotPresentInYaml",
+			args: args{
+				field: "environment",
+			},
+			want: "<invalid Value>",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := GetDefaultValueByStringField(tt.args.field); got != tt.want {
+				t.Errorf("GetDefaultValueByStringField() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestGetUserConfValueByStringField(t *testing.T) {
+	type args struct {
+		field string
+	}
+	// Set Value of Conf
+	Conf = Configuration{
+		Environment: "minikube",
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "FieldPresentInConf",
+			args: args{
+				field: "Environment",
+			},
+			want: "minikube",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := GetUserConfValueByStringField(tt.args.field); got != tt.want {
+				t.Errorf("GetUserConfValueByStringField() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+	Conf = Configuration{} // Reset value of Conf to default
+}
+
+func TestGetConf(t *testing.T) {
+	os.Setenv("CITF_CONF_ENVIRONMENT", "minikube")
+
+	Conf = Configuration{
+		Environment: "Dear minikube",
+	}
+
+	type args struct {
+		field string
+	}
+
+	tests := []struct {
+		name string
+		args args
+		want string
```

Comment:
Just a thought here. can we have a `cleanup func()` here in the struct which will have all cleanup code for that particular test case. We'll just call `cleanup()` for every test in the loop below. This way we can bind the cleanup code to the test itself.
Also we can avoid all the conditions matching below (which will soon be non-maintainable also if test grows).